storage/shard.go

```diff
@@ -424,9 +427,9 @@ func (s *dbShard) Bootstrap(
 
 	multiErr := xerrors.NewMultiError()
 	for _, dbBlocks := range bootstrappedSeries {
-		series, _, completionFn := s.writableSeries(dbBlocks.ID)
-		err := series.Bootstrap(dbBlocks.Blocks)
-		completionFn()
+		entry := s.writableSeries(dbBlocks.ID)
+		err := entry.series.Bootstrap(dbBlocks.Blocks)
+		entry.decrementWriterCount()
```

Comment:
Perhaps on `entry` we should add a method called `beginWriteOp(...)` that returns the entry itself after incrementing the writer count and then a wrapper for decrement writer count too:
```go
func (e *dbShardEntry) beginWriteOp() *dbShardEntry {
  e.incrementWriterCount()
  return e
}

func (e *dbShardEntry) endWriteOp() *dbShardEntry {
  e.decrementWriterCount()
  return e
}
```

That way we can write this as:
```go
entry := s.writableSeries(dbBlocks.ID).beginWriteOp()
err := entry.series.Bootstrap(dbBlocks.Blocks)
entry.endWriteOp()
```

Which is much more sane.  I'm sure the functions for increment/decrement would get inlined into the two syntactic methods.